src/AutoMapper/MappingEngine.cs

```diff
@@ -212,28 +190,22 @@ public TDestination DynamicMap<TDestination>(object source)
 
         public object DynamicMap(object source, Type sourceType, Type destinationType)
         {
+            ConfigurationProvider.CreateMissingTypeMaps = true;

```

Comment:
And I think I might be missing what your suggestion is. Maybe I'll just merge it and you can give it a go? Part of the purpose here is to have those object mappers only match on a TypePair.